How to create a headless CMS website using WordPress, Next.js and GraphQL Queries (WPGraphQL)

Catemacharia opened the pull request:
### Article checklist

#### Pre-submission checks
> Please ensure that you have only one open topic suggestion (issue) + in-review article (linked pull request) at a time. This will ensure that we complete the review process in a timely manner from inception to publishing

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

#### Topic suggestion this closes
> Remove the backticks and add the issue number below to link and close your Topic Suggestion (issue) when your article has been published (PR has been merged). See this [video](https://youtu.be/TKJ4RdhyB5Y?t=278) for more details.

This closes #3098

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

Bayler commented:
Hi @Catemacharia You have unchecked boxes on your article checklist. Please note that all the items in the article checklist need to be followed and checked.

WanjaMIKE commented:
Plagiarism check done - Proceed.

Bayler commented:
Hi @Catemacharia You have unchecked boxes on your article checklist. Please note that all the items in the article checklist need to be followed and checked. Also, please add an introduction section above the `Overview` section.

Catemacharia commented:
> Hi @Catemacharia You have unchecked boxes on your article checklist. Please note that all the items in the article checklist need to be followed and checked. Also, please add an introduction section above the `Overview` section.

Doing that ASAP @Bayler

Bayler commented:
@Catemacharia Are you still working on them? It's taking too long

Catemacharia commented:
> @Catemacharia Are you still working on them? It's taking too long

Working on it right now @Bayler

Catemacharia commented:
> @Catemacharia Are you still working on them? It's taking too long

@Bayler Check the introductory section, I have added a few lines.

Bayler commented:
> > @Catemacharia Are you still working on them? It's taking too long
> 
> @Bayler Check the introductory section, I have added a few lines.

You still haven't checked all the checkboxes

Bayler commented:
> > > @Catemacharia Are you still working on them? It's taking too long
> > 
> > 
> > @Bayler Check the introductory section, I have added a few lines.
> 
> You still haven't checked all the check boxes. Let's not spend a lot of time tackling these basics.

Catemacharia commented:
> > > > @Catemacharia Are you still working on them? It's taking too long
> > > 
> > > 
> > > @Bayler Check the introductory section, I have added a few lines.
> > 
> > 
> > You still haven't checked all the check boxes. Let's not spend a lot of time tackling these basics.

@Bayler , All images are screenshots that I took. I have no outsourced images, that's why I didn't check that box,

Bayler commented:
> > > > > @Catemacharia Are you still working on them? It's taking too long
> > > > 
> > > > 
> > > > @Bayler Check the introductory section, I have added a few lines.
> > > 
> > > 
> > > You still haven't checked all the check boxes. Let's not spend a lot of time tackling these basics.
> 
> @Bayler , All images are screenshots that I took. I have no outsourced images, that's why I didn't check that box,

I understand. Nevertheless, all check boxes need to be checked.

Bayler commented on content/articles/create-a-headless-cms-website-using-wordpress-nextjs-and-graphql/index.md:
```diff
@@ -0,0 +1,439 @@
+- [Testing](#testing)
+- [Testing with a single post](#testing-with-a-single-post)
+- [Conclusion](#conclusion)
+- [References](#references)
+
+### What is a headless CMS
```
Any prerequisites needed for a reader to understand this tutorial? Please include a prerequisite section

Bayler commented on content/articles/create-a-headless-cms-website-using-wordpress-nextjs-and-graphql/index.md:
```diff
@@ -0,0 +1,439 @@
+
+- API driven data promote omnichannel architectures. You create WordPress content and channel it through API. Thus the content shines across all touchpoints.
+
+- It creates more efficient, scalable, and fast applications. Both backend and frontend are separated from the monolith CMS architecture. This means if the WordPress-backed CMS has any issues, the channel delivery pipelines are not affected. Hence the performance of the delivery channel is not compromised.
+
+### Adding a post
+
```
Please include a paragraph joining these two sections. The tutorial should flow from one section to another

Bayler commented on content/articles/create-a-headless-cms-website-using-wordpress-nextjs-and-graphql/index.md:
```diff
@@ -0,0 +1,439 @@
+### Adding a post
+
+Since WordPress and Next.js are used to develop blog-based websites, we will use this scenario by first adding and posting a post in WordPress admin.
+
+First, from your WpAdmin dashboard, navigate to `posts` and add a `category`. On the left pane of the resulting page, enter a name, slug, and a description for the post category, and then click the button `Add new category`. Here is how I entered these fields.
+
```
A reader not familiar with WordPress would wonder how does one get to the admin dashboard in the first place. You can include a link or explain how one gets to the WpAdmin dashboard

Bayler commented on content/articles/create-a-headless-cms-website-using-wordpress-nextjs-and-graphql/index.md:
```diff
@@ -0,0 +1,439 @@
+From the left sidebar of the dashboard, Navigate to plugins. On the resulting pop-up to the right, click on `Add new`. On the search bar, search `WpGraphQL`:
+
+![plugins-search-bar](/engineering-education/create-a-headless-cms-website-using-wordpress-nextjs-and-graphql/plugins-search-bar.png).
+
+Click `Install now` on the following result. Once the installation is over, make sure you activate the plugin.
+
```
You could add a screenshot on how to activate the plugin

Bayler commented on content/articles/create-a-headless-cms-website-using-wordpress-nextjs-and-graphql/index.md:
````diff
@@ -0,0 +1,439 @@
+  }
+ }
+```
+
+We are fetching all posts from the above query, with each getting the post title, category, excerpt, and content. On the left pane, paste in the above query. Then, execute the query by hitting the play button.
````
You could also add a screenshot here with the play button

Bayler commented on content/articles/create-a-headless-cms-website-using-wordpress-nextjs-and-graphql/index.md:
````diff
@@ -0,0 +1,439 @@
+```bash
+npx create-next-app cms-wordpress-app
+```
+
+- A folder `cms-wordpress-app` will be created, which will contain all the necessary files to start working on a Next.js project. Change to the directory to this folder path (project folder) by running;
+
````
```
- A folder `cms-wordpress-app` will be created, which will contain all the necessary files to start working on a Next.js project. Use the Change Directory (cd) command to get into the project folder `cms-wordpress-app` we just created;

Bayler commented on content/articles/create-a-headless-cms-website-using-wordpress-nextjs-and-graphql/index.md:
```diff
@@ -0,0 +1,439 @@
+- Go to your WordPress admin page.
+
+- On the left sidebar, click on GraphQL.
+
+- Get your URL under *GraphQL Endpoint*.
+
```
Please include a screenshot

Bayler commented on content/articles/create-a-headless-cms-website-using-wordpress-nextjs-and-graphql/index.md:
````diff
@@ -0,0 +1,439 @@
+```
+
+In your browser, navigate to `http://localhost:3000`.
+
+To test the implementation, click on the title of any post on the home page. You should be redirected to a single post page. Then, click on the back arrow and explore the other posts.
+
````
Please include a screenshot of the result obtain when you run that URL on the browser

Bayler commented on content/articles/create-a-headless-cms-website-using-wordpress-nextjs-and-graphql/index.md:
```diff
@@ -0,0 +1,439 @@
+
+To test the implementation, click on the title of any post on the home page. You should be redirected to a single post page. Then, click on the back arrow and explore the other posts.
+
+### Conclusion
+We have managed to create posts in WordPress, query, and show them from a Next.js application. To gain more insightful knowledge about the various technologies and concepts used, check out the following resources.
```
Here, you mention that the reader should check out the following resources, do you mean the resources on the reference section? If so, it would be able to instead write, "...check out the resources on the reference section"

Bayler commented:
Hi @Catemacharia This is a great tutorial. I've really learned a lot reviewing it. Just a few corrections then we can move on to final review.

Catemacharia commented:
@Bayler Check out the new changes. I have updated them to reflect the few suggested corrections.

Catemacharia commented:
> @Bayler Check out the new changes. I have updated them to reflect the few suggested corrections.
 @Bayler Are all changes okay?

Bayler commented:
> > @Bayler Check out the new changes. I have updated them to reflect the few suggested corrections.
> 
> @Bayler Are all changes okay?

Please rename the file to wp-graphql-playbutton.png

create-a-headless-cms-website-using-wordpress-nextjs-and-graphql/wp-graphql-playbatton.png

Catemacharia commented:
> > > @Bayler Check out the new changes. I have updated them to reflect the few suggested corrections.
> > 
> > 
> > @Bayler Are all changes okay?
> 
> Please rename the file to wp-graphql-playbutton.png
> 
> create-a-headless-cms-website-using-wordpress-nextjs-and-graphql/wp-graphql-playbatton.png

I have updated the file @Bayler

hectorkambow commented:
Overall great work 🚀
